Add Fleet & Agent 8.14.0 Release Notes (#1079)
elastic/ingest-docs: docs/en/ingest-management/release-notes/release-notes-8.14.asciidoc

{fleet-server}::
* When running in `agent` mode, {fleet-server} will use the APMConfig settings of the expected input if it's set over the settings in `inputs[0].server.instrumentation`. This should make it easier for managing agents to inject APM configuration data. {fleet-server-pull}3277[3277] {fleet-server-issue}2868[2868]
* Allow specification in the {fleet-server} settings for whether or not a diagnostics bundle should contain additional CPU metrics. {fleet-server-pull}3333[3333] {agent-issue}3491[3491]
* Allow {fleet} to set the trace level for logging. {fleet-server-pull}3350[3350]

{agent}::
* Change the behavior of the monitoring {beats} so they also monitor and report metrics on themselves. {agent-pull}4326[#4326] {agent-issue}4082[#4082]
* Add the optional CPU profile collection to the {fleet} diagnostics action handler. {agent-pull}4394[#4394] {agent-issue}3491[#3491]
* Enable `--unprivileged` on Mac OS, allowing {agent} to run as an unprivileged user. {agent-pull}4362[#4362] {agent-issue}3867[#3867]
* Make the `enroll` command more stable by handling temporary server errors. {agent-pull}4523[#4523] {agent-issue}4513[#4513]

> * Reduce the overall download and on-disk size of {agent}. {agent-pull}4516[#4516] {agent-issue}3364[#3364]

Reviewer: Updated with the sizes. I suggested sub-bullets but I'm not sure if that works here, change the formatting as needed.

> * Reduce the overall download and on-disk size of {agent}. {agent-pull}4516[#4516] {agent-issue}3364[#3364]

Author: @cmacknz  Thanks a lot for providing the exact text in all these suggestions!

I updated this one via [db6d149](https://github.com/elastic/ingest-docs/pull/1079/commits/db6d149728db1b610c9ca30fb36da382d5de72e1) to make sure it renders nicely:

![Screenshot 2024-05-22 at 2 39 41 PM](https://github.com/elastic/ingest-docs/assets/41695641/794bd920-f792-46cf-ad3a-5537ecfa87d7)

:agent-libs-pull: https://github.com/elastic/elastic-agent-libs/pull/
:agent-issue: https://github.com/elastic/elastic-agent/issues/
:agent-pull: https://github.com/elastic/elastic-agent/pull/
:fleet-server-issue: https://github.com/elastic/fleet-server/issues/
:fleet-server-pull: https://github.com/elastic/fleet-server/pull/

[[release-notes]]
= Release notes

This section summarizes the changes in each release.

* <<release-notes-8.14.0>>

Also see:

* {kibana-ref}/release-notes.html[{kib} release notes]
* {beats-ref}/release-notes.html[{beats} release notes]

// begin 8.14.0 relnotes

[[release-notes-8.14.0]]
== {fleet} and {agent} 8.14.0

Review important information about {fleet-server} and {agent} for the 8.14.0 release.

[discrete]
[[security-updates-8.14.0]]
=== Security updates

{fleet-server}::
* Update {fleet-server} Go version to 1.21.10. {fleet-server-pull}3528[#3528]

{agent}::
* Update {agent} Go version to 1.21.10. {agent-pull}4718[#4718]
* Update all `opentelemetry-collector-contrib` packages. {agent-pull}4572[#4572]

[discrete]
[[new-features-8.14.0]]
=== New features

The 8.14.0 release added the following new and notable features.

{fleet}::

> * Adds subfeatures privileges for {fleet}, {agents}, agent policies, and settings, this feature is in technical preview. ({kibana-pull}179889[#179889]).

Reviewer 2: Not sure if there's a more formal way to include a technical preview badge or something here

* Avoids subobject and scalar mapping conflicts by setting `subobjects: false` on custom integrations. ({kibana-pull}178397[#178397])
* Adds functionality to `default_fields` field, so a query can run against all fields in the mapping. ({kibana-pull}178020[#178020])
* Relaxes delete restrictions for managed content installed by {fleet}. ({kibana-pull}179113[#179113])

{agent}::
* The Kubernetes secrets provider has been improved to update a Kubernetes secret  when the secret value changes. {agent-pull}4371[#4371] {agent-issue}4168[#4168]
* The OpenTelemetry link:https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/filterprocessor[filterprocessor] is now available to users running {agent} in `otel` mode. {agent-pull}4708[#4708]
* The OpenTelemetry link:https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/elasticsearchexporter[elasticsearchexporter] is now available to users running {agent} in `otel` mode. {agent-pull}4707[#4707]

[discrete]
[[enhancements-8.14.0]]
=== Enhancements

{fleet}::
* Adds support for dimension mappings in dynamic templates. ({kibana-pull}180023[#180023])

> * Adds CPU metrics to request diagnostics. ({kibana-pull}179819[#179819])

Reviewer 2: ```suggestion
* Allow additional CPU metrics to be collected when requesting diagnostics from an agent. ({kibana-pull}179819[#179819])
```

> * Adds Settings Framework API and UI. ({kibana-pull}179795[#179795])

Reviewer 2: ```suggestion
* Add new "advanced settings" section to agent policy settings page sourced from configuration. ({kibana-pull}179795[#179795])
```

> * Adds an Elastic Defend advanced policy option for pruning capability arrays. ({kibana-pull}179766[#179766])

Reviewer 2: ```suggestion
* Add an Elastic Defend advanced policy option for pruning capability arrays. ({kibana-pull}179766[#179766])
```

> * Adds {agent} activity flyout enhancements. ({kibana-pull}179161[#179161])

Reviewer 2: ```suggestion
* The "agent activity" flyout now includes several new features. ({kibana-pull}179161[#179161])
  ** A "review errors" button now appears above the agent listing table when new activity events are loaded that include errors. Clicking the button will open the activity flyout with these errors shown.
  ** Agent activity now supports pagination. Click the "show more" button at the bottom of the list to load additional activity events.
  ** Agent activity from a given date can now be loaded by clicking the "Go to date" button and selecting a date. 
```

Tried to mimic the sub-bullets that appear below here. Hope I got the formatting right!

> * Adds unhealthy reason (input/output/other) to agent metrics. ({kibana-pull}178605[#178605])

Reviewer 2: ```suggestion
* Surface `unhealthy_reason` in agent metrics that indicates which component (input/output/other) is causing an agent to be considered unhealthy. ({kibana-pull}178605[#178605])
```

> * Adds a warning which is displayed when trying to upgrade agent to version > max {fleet-server} version. ({kibana-pull}178079[#178079])

Reviewer 2: ```suggestion
* Add a warning which is displayed when trying to upgrade agent to version > max {fleet-server} version. ({kibana-pull}178079[#178079])
```

* Add the optional CPU profile collection to the {fleet} diagnostics action handler. {agent-pull}4394[#4394] {agent-issue}3491[#3491]
* Enable `--unprivileged` on Mac OS, allowing {agent} to run as an unprivileged user. {agent-pull}4362[#4362] {agent-issue}3867[#3867]
* Make the `enroll` command more stable by handling temporary server errors. {agent-pull}4523[#4523] {agent-issue}4513[#4513]
* Reduce the overall download and on-disk size of {agent}. {agent-pull}4516[#4516] {agent-issue}3364[#3364]
** Linux: -43% reduction from 1800MB to 1018MB compared to 8.13.4 when extracted
** MacOS: -44% reduction from 1100MB to 619MB compared to 8.13.4 when extracted
** Windows: -43% reduction from 891MB to 504MB compared to 8.13.4 when extracted
* Remove `cloud-defend` from Linux `.tar.gz` archives; it now appears only in Docker images where it is required. {agent-pull}4584[#4584]
* Reduce the disk usage of {agent} self-monitoring logs shipped to {fleet} by 16% by dropping "Non-zero metrics..." logs automatically. {agent-pull}4633[#4633] {agent-issue}4252[#4252]

[discrete]
[[bug-fixes-8.14.0]]
=== Bug fixes

{fleet}::

> * Adds validation to dataset field in input packages to disallow special characters. ({kibana-pull}182925[#182925])

Reviewer 2: ```suggestion
* Add validation to dataset field in input packages to disallow special characters. ({kibana-pull}182925[#182925])
```

> * Fixes rollback input package install on failure. ({kibana-pull}182665[#182665])

Reviewer 2: ```suggestion
* Fix rollback input package install on failure. ({kibana-pull}182665[#182665])
```

> * Fixes cloudflare template error. ({kibana-pull}182645[#182645])

Reviewer 2: ```suggestion
* Fix cloudflare template error. ({kibana-pull}182645[#182645])
```

> * Fixes displaying `Config` and `API reference` tabs if they are not needed. ({kibana-pull}182518[#182518])

Reviewer 2: ```suggestion
* Fix displaying `Config` and `API reference` tabs if they are not needed. ({kibana-pull}182518[#182518])
```

* Allow upgrading an agent to a newer version when that agent is also a {fleet-server}. ({kibana-pull}181575[#181575])

> * Fixes flattened inputs in the configuration tab. ({kibana-pull}181155[#181155])

Reviewer 2: ```suggestion
* Fix flattened inputs in the configuration tab. ({kibana-pull}181155[#181155])
```

> * Adds callout when editing an output about plain text secrets being re-saved to secret storage. ({kibana-pull}180334[#180334])

Reviewer 2: ```suggestion
* Add callout when editing an output about plain text secrets being re-saved to secret storage. ({kibana-pull}180334[#180334])
```

> * Removes unnecessary field definitions for custom integrations. ({kibana-pull}178293[#178293])

Reviewer 2: ```suggestion
* Remove unnecessary field definitions for custom integrations. ({kibana-pull}178293[#178293])
```

> * Fixes secrets UI inputs in forms when secrets storage is disabled server side. ({kibana-pull}178045[#178045])

Reviewer 2: ```suggestion
* Fix secrets UI inputs in forms when secrets storage is disabled server side. ({kibana-pull}178045[#178045])
```

> * Fixes not being able to preview or download files with special characters. ({kibana-pull}176822[#176822])

Reviewer 2: ```suggestion
* Fix not being able to preview or download files with special characters. ({kibana-pull}176822[#176822])
```

> * Fixes KQL validation being applied in search boxes. ({kibana-pull}176806[#176806])

Reviewer 2: ```suggestion
* Fix overly strict KQL validation being applied in search boxes. ({kibana-pull}176806[#176806])
```
